src/Nethermind/Nethermind.Synchronization/StateSync/StateSyncDownloader.cs

```diff
@@ -65,7 +65,7 @@ public async Task Dispatch(PeerInfo peerInfo, StateSyncBatch batch, Cancellation
                 else
                 {
                     if (Logger.IsTrace) Logger.Trace($"Requested TrieNodes via SnapProtocol from peer {peer}");
-                    GetTrieNodesRequest request = GetGroupedRequest(batch);
+                    using GetTrieNodesRequest request = GetGroupedRequest(batch);
```

Comment:
this will be disposed at the end of this request, when task can still be in progress?

Reply:
moved dispose after awaiting on task result